tests/unit/test_hnsw_tiered.cpp

```diff
@@ -0,0 +1,79 @@
+#include "VecSim/algorithms/hnsw/hnsw_tiered.h"
+#include "VecSim/algorithms/hnsw/hnsw_factory.h"
+#include "test_utils.h"
+
+using namespace tiered_index_mock;
+
+template <typename index_type_t>
+class HNSWTieredIndexTest : public ::testing::Test {};
+
+TYPED_TEST_SUITE(HNSWTieredIndexTest, DataTypeSet);
+
+TYPED_TEST(HNSWTieredIndexTest, CreateIndexInstance) {
+    std::shared_ptr<VecSimAllocator> allocator = VecSimAllocator::newVecsimAllocator();
+
+    // Create TieredHNSW index instance with a mock queue.
+    for (auto is_multi : {true, false}) {
+        HNSWParams params = {.type = TypeParam::get_index_type(),
+                             .dim = 4,
+                             .metric = VecSimMetric_L2,
+                             .multi = is_multi};
+        auto *jobQ = new JobQueue();
+        size_t memory_ctx = 0;
+        TieredIndexParams tiered_params = {.jobQueue = jobQ,
+                                           .submitCb = submit_callback,
+                                           .memoryCtx = &memory_ctx,
+                                           .UpdateMemCb = update_mem_callback};
+        TieredHNSWParams tiered_hnsw_params = {.hnswParams = params, .tieredParams = tiered_params};
+        auto *tiered_index = reinterpret_cast<TieredHNSWIndex<TEST_DATA_T, TEST_DIST_T> *>(
+            HNSWFactory::NewTieredIndex(&tiered_hnsw_params, allocator));
+
+        // Add a vector to the flat index.
+        TEST_DATA_T vector[tiered_index->index->getDim()];
+        GenerateVector<TEST_DATA_T>(vector, tiered_index->index->getDim());
+        labelType vector_label = 1;
+        VecSimIndex_AddVector(tiered_index->tempFlat, vector, vector_label);
+
+        // Create a mock job that inserts some vector into the HNSW index.
+        auto insert_to_index = [](void *job) {
+            auto *my_insert_job = reinterpret_cast<HNSWInsertJob *>(job);
+            auto my_index =
+                reinterpret_cast<TieredHNSWIndex<TEST_DATA_T, TEST_DIST_T> *>(my_insert_job->index);
+
+            // Move the vector from the temp flat index into the HNSW index.
+            my_index->addVector(my_index->tempFlat->getDataByInternalId(my_insert_job->id),
+                                my_insert_job->label);
+            VecSimIndex_DeleteVector(my_index->tempFlat, my_insert_job->label);
+            auto it = my_index->labelToInsertJobs[my_insert_job->label].begin();
```

Comment:
explain this flow happens in a fixture that expects this task already be in the queue
in other words
this fixture tests the `labelToInsertJobs` correctness